src/app-error.ts

```diff
@@ -0,0 +1,57 @@
+import os from 'os';
+import { formatPercent } from './helpers';
+import { ErrorCode } from './index';
+
+// If we need advanced error consider using https://github.com/joyent/node-verror
+export class AppError extends Error {
+  code?: string;
```

Comment:
`string` → `ErrorCode`?

I generally use unions of string literals instead of enums, but curious to hear your thoughts. See https://medium.com/@katbusch/typescript-enums-explained-e5f9a101afc9

Reply:
No advantages of using enum here. I will replace it by string literal type

Reply:
Replaced by string literal type